File: examples/projects/benchmarks/actor_pingpong/actor_pingpong.go
// 基准: actor_pingpong —— 5 万次 goroutine 间同步往返（双无缓冲 channel）
// 与 actor_pingpong.rl 逻辑严格一致。输出 = 1250025000
package main

import "fmt"

const n = 50000

func worker(slotA <-chan int64, slotB chan<- int64) {
	var count int64
	for i := 0; i < n; i++ {
		<-slotA
		count++
		slotB <- count
	}
}

func main() {
	slotA := make(chan int64)
	slotB := make(chan int64)
	go worker(slotA, slotB)
	var sum int64
	for i := int64(0); i < n; i++ {
		slotA <- i
		sum += <-slotB
	}
	fmt.Println(sum)
}

File: examples/projects/benchmarks/actor_pingpong/actor_pingpong.swift
// 基准: actor_pingpong —— 5 万次线程间同步往返（NSCondition 双槽握手）
// 与 actor_pingpong.rl 逻辑严格一致。输出 = 1250025000
import Foundation

let N = 50000
let cond = NSCondition()
var slotAReady = false, slotBReady = false
var slotA: Int64 = 0, slotB: Int64 = 0

let worker = Thread {
    var count: Int64 = 0
    for _ in 0..<N {
        cond.lock()
        while !slotAReady { cond.wait() }
        slotAReady = false
        count += 1
        slotB = count
        slotBReady = true
        cond.signal()
        cond.unlock()
    }
}
worker.start()

var sum: Int64 = 0
for i in 0..<N {
    cond.lock()
    slotA = Int64(i)
    slotAReady = true
    cond.signal()
    while !slotBReady { cond.wait() }
    sum += slotB
    slotBReady = false
    cond.unlock()
}
print(sum)
